cmake/public/utils.cmake

```diff
@@ -249,7 +247,7 @@ function(torch_compile_options libname)
         /bigobj
         )
     else()
-      target_compile_options(${libname} PUBLIC
+      target_compile_options(${libname} PRIVATE
         #    -std=c++14
         -Wall
         -Wextra
```

Comment:
I still think we shouldn't turn off all the them, at least for MSVC. It is very common that a library expose some flags for the compiler and the linker. We shouldn't stop to pass those args just because they cannot be correctly propagated in the newer versions of CMake with the cuda language support.

Reply:
I wrote why I don't think that's ideal in #36021, but if msvc really needs to be warnings-free, I will drop the msvc part.